src/main/java/at/hannibal2/skyhanni/data/MayorAPI.kt

```diff
@@ -0,0 +1,131 @@
+package at.hannibal2.skyhanni.data
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.config.ConfigManager
+import at.hannibal2.skyhanni.data.jsonobjects.local.MayorJson
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.utils.APIUtil
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.put
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SimpleTimeMark.Companion.asTimeMark
+import io.github.moulberry.notenoughupdates.util.SkyBlockTime
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.minutes
+
+object MayorAPI {
+    private var lastUpdate = SimpleTimeMark.farPast()
+    private var dispatcher = Dispatchers.IO
+
+    var rawMayorData: MayorJson? = null
+        private set
+    var candidates = mapOf<Int, MayorJson.Candidate>()
+        private set
+    var currentMayor: MayorJson.Candidate? = null
+        private set
+    var timeTillNextMayor = Duration.ZERO
+        private set
+
+    private const val LATE_SPRING = 3
+
+    fun isPerkActive(mayor: String, perk: String) = currentMayor?.let { currentCandidate ->
+        currentCandidate.name == mayor && currentCandidate.perks.any { it.name == perk }
+    } ?: false
+
+    /**
+     * @param input: The name of the mayor
+     * @return: The neu color of the mayor; If no mayor was found, it will return "§cUnknown: §7"
+     */
+    fun mayorNameToColorCode(input: String): String {
+        return when (input) {
+            // Normal Mayors
+            "Aatrox" -> "§3"
+            "Cole" -> "§e"
+            "Diana" -> "§2"
+            "Diaz" -> "§6"
+            "Finnegan" -> "§c"
+            "Foxy" -> "§d"
+            "Marina" -> "§b"
+            "Paul" -> "§c"
+
+            // Special Mayors
+            "Scorpius" -> "§d"
+            "Jerry" -> "§d"
+            "Derpy" -> "§d"
+            "Dante" -> "§d"
+            else -> "§cUnknown: §7"
+        }
+    }
```

Comment:
Maybe this could be setup as an enum

Reply:
Yeah, we probably should have a mayor enum. Could also make the perk checking simpler. And an enum should be good here since Hypixel is unlikely to add or change a mayor (maybe in the foraging update 🙃).

Reply:
Surely its good like this?

Reply:
oooh rightt

Reply:
Should be fixed